frontend/src/components/PipelinesDialogV2.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import { classes } from 'typestyle';
+import { padding, commonCss } from 'src/Css';
+import DialogContent from '@material-ui/core/DialogContent';
+import ResourceSelectorV2 from 'src/pages/ResourceSelectorV2';
+import { Apis, PipelineSortKeys } from 'src/lib/Apis';
+import { Column } from './CustomTable';
+import { V2beta1Pipeline } from 'src/apisv2beta1/pipeline';
```

Comment:
The reason I choose to separate v1 and v2 here is to avoid changes on v1 implementation. We can definitely merge the logic together, but I am not sure if changing v1 code base (cause some risks of regression) is preferred at this moment.